swarm/api/api.go

```diff
@@ -40,6 +42,66 @@ type Resolver interface {
 	Resolve(string) (common.Hash, error)
 }
 
+// errNoResolver is returned by MultiResolver.Resolve if no resolver
+// can be found for the address.
+var errNoResolver = errors.New("no resolver")
+
+// MultiResolver is used to resolve URL addresses based on their TLDs.
+// Each TLD can have multiple resolvers, and the resoluton from the
+// first one in the sequence will be returned.
+type MultiResolver struct {
+	resolvers map[string][]Resolver
+}
+
```

Comment:
why do we need multiple resolvers per TLD is only the frist resolution is returned?

Reply:
As per requirement in this comment https://github.com/ethereum/go-ethereum/issues/14386#issuecomment-334378961 for sequence of resolvers.